nats/bench_command.go

```diff
@@ -159,7 +159,7 @@ func (c *benchCmd) bench(_ *kingpin.ParseContext) error {
 		}
 
 		// First delete any prior stream unless no delete
-		if !c.noDeleteStream {
+		if c.deleteStream {
```

Comment:
so part of deleting it is that it's safe to do this

https://github.com/nats-io/natscli/blob/57a26979704d591774f0deeaba412e120f2936c8/nats/bench_command.go#L184-L185

What we cant have is if someone gives a stream name that we then go and edit the properties of that stream, if they give a stream name and it exist, we either have to somehow adapt to it and bench against it or something? 

I don't know, to be honest the ability to take arbitrary stream names seems like a trap to me.

When we create the consumer, there's no filter subjects - so again on a pre-existing stream this will go bad. So we'd need to ensure we filter only on the given c.subject

I am more and more unconvinced about the ability to operate on anything but a clean environment ie. stream must not exist before we start.  We should never `make sure the existing one has the same attributes`.

Reply:
Sure for consumption obviously the stream must exist. But what about publishing is it not a safe approach to say the stream must not be there? Anyway as you clarify the client will fail for differences in config so it’s probsbly fine. 

Sounds good we can take either approach with delete or not etc - I am not trying to dictate the design I just want to be sure it’s safe for users if they point it at a stream with prod data that’s all

Reply:
So I went with removing the --delete functionality altogether (and improving the error messages for the cases where the user may need to manually delete the stream first). Telling them to use `nats stream rm` (rather than deleting directly) also means they get prompted "are you sure?" before the delete actually happens.